### DISCUSSION TITLE
<unclear what is being asked>

### DISCUSSION THREAD

**Discussion by jimlynnjulian** [BurntSushi/ripgrep#2594] [General] [answered]
There does not seem to be a way to catch the output of ripgrep in a textfile.
My script is:
``# rg script
date
rg.exe -l "KaxSeekHead" -g '*.txt' >> _rip-out-Aeschylus.txt
#rg.exe -l "KaxSeekHead" -g '*.txt' 2>&1 | tee -a _rip-out-Aeschylus.txt
#rg.exe -l "KaxSeekHead" -g '*.txt' | tee -a _rip-out-Aeschylus.txt
#rg.exe -l "KaxSeekHead" -g '*.txt'
date
``
All fail.  The program takes about two seconds without the pipe. With the pipe, the program seems to hang.
Can the output be directed to a text file?
Thank you.
Jim J.

**Comment by BurntSushi** [comment]
Sounds to me like you're searching the same file that you're writing to, creating an infinite loop. Why not check the contents of the file to see if anything has been written?

(ripgrep has no problem being piped to a file or another program. You might try a little debugging by trying a simpler example to confirm.)

**Reply by jimlynnjulian** [reply]
For clarity:
My script is:

# rg script 

date 

rg.exe -l "KaxSeekHead" -g '*.txt' >> _rip-out-Aeschylus.txt 


#rg.exe -l "KaxSeekHead" -g '*.txt' 2>&1 | tee -a _rip-out-Aeschylus.txt 

#rg.exe -l "KaxSeekHead" -g '*.txt' | tee -a _rip-out-Aeschylus.txt 

#rg.exe -l "KaxSeekHead" -g '*.txt' 

date

**Reply by BurntSushi** [reply]
Yes, you're searching a directory while also writing the results of that search to a file in the same directory.....

**Reply by jimlynnjulian** [reply, marked as answer]
Just ran the script successfully. The problem is Windows WSL. I've noticed that occasionally, things will start to go wrong while working in bash. Simply closing the window doesn't work. The computer has to be re-booted. That's what I just finished doing before running the script again. Thanks for your input.
Additionally, I was using the Windows exec in the system path. When I installed using apt install instead, everything worked.

**Comment by jimlynnjulian** [comment]
Also, I use two laptops. One is Win10Pro, the other is Win11Pro. When I tried to install the linux version on the later, A suitable version could not be found. Then I installed the Windows exec on system path. This added to the confusion.
The website stipulates the linux (Ubuntu) version is limited, hence the lack of a linux install.
The two versions are not completely compatible. One accepts 'rg' on the command line, the other requires 'rg.exe.'
Fortunately, persistence has been rewarded.
Again, thank you for your time.

Oh, yes, the question was 'can the output of rg be piped to a file', yes is your answer. Nothing unclear about that.

**Reply by BurntSushi** [reply]
Yeah I can't really make heads or tails of what you're saying. But it sounds like you got it working. That's great.

**Comment by jimlynnjulian** [comment]
Thanks. I'll try to be more succinct. 
One laptop had Win11 & rg.exe, the other had Win10 & rg(linux-Ubuntu).
The exe version was not compatible with my script. The linux version was. 
This made writing and testing the script on both machines confusing.
The reaon I used the exe version is the WSL Ubuntu version disallowed ripgrep installation
due lack of availablility of a qualifying version.
I hope this clarifies things some. Anyway...

Thanks again.
